lib/swoc/include/swoc/IPRange.h

```diff
@@ -1145,22 +1206,85 @@ class IPRangeSet
   /// @return Number of ranges in the set.
   size_t count() const;
 
+  bool empty() const;
+
   /// Remove all addresses in the set.
   void clear();
 
-protected:
-  /// Empty struct to use for payload.
-  /// This declares the struct and defines the singleton instance used.
-  static inline constexpr struct Mark {
-    using self_type = Mark;
-    /// @internal @c IPSpace requires equality / inequality operators.
-    /// These make all instance equal to each other.
-    bool operator==(self_type const &that);
-    bool operator!=(self_type const &that);
-  } MARK{};
+  /// Constant iterator for iteration over ranges.
+  class const_iterator {
```

Comment:
Say what type or the concepts:  https://en.cppreference.com/w/cpp/iterator

Reply:
My understanding is that inheriting from iterator tags is deprecated and instead the types should be provided via `using`.

Reply:
OK, I'll put an update in libswoc.

Reply:
So we're manually copying libswoc code and checking it into TS? I wonder if we could implement this logic in the make file:
```
LIBSWOC_VERSION=yada
if [[ ! -d libswoc || ! -f libswoc/version || $LIBSWOC != $(cat libswoc/version) ]] ; then
    rm -rf libswoc
    git clone something-something libswoc
    echo $LIBSWOC_VERSION > libswoc/version # even better if libswoc alread has this file
fi
```
and also be non-Damian and update .gitignore to ignore these source files.

Reply:
It was decided to do it this way because that's how the YAML library is handled.  You can look [here](https://github.com/SolidWallOfCode/libswoc/blob/master/tools/ats-drop.sh) for how the code is copied. That could be move over to ATS and then invoked from a makefile.